python/tank/authentication/shotgun_shared/authentication_session_data.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2017 Autodesk.
```

Comment:
could this be more indepdent and handled like the shotgun API? E.g. here: `https://github.com/shotgunsoftware/tk-core/tree/master/python/tank_vendor`? What's the workflow for upgrading this code in toolkit from the 'common' repo where it is stored? Feels like this would be worth discussion!

Reply:
is there already another repo where the source of this code lives? With tests etc? Would be great to discuss what that looks like! Ideally, i think we want a sample app/reference implementation which uses this module, so that we can develop and QA it in isolation from toolkit. This would also allow (future) clients who aren't using toolkit (but using the sg api) to create tools which support SSO.

Reply:
i hate to create more work, but i think when we discussed it a while back, we perhaps weren't quite ready to have a separate repo - if we want this to be a truly shared component between rv, core and any other SSO client of the SG API it would be great if there was a repo for it, with some readme, a simple 'hello world' type example and tests. And a version number. It would make it consistent with how we use the sg API in toolkit but i think also it would help making any future updates easier - and of course good for any client who wants to use SSO with their custom tools running outside RV/tk.

Reply:
I understand and totally agree with you.

I have created a new GitHub Repo, called saml2_sso. But it will take a bit of work to make it pretty (it is a private repo at the moment, but  I plan to make it public early in January once I am happy with the sample, and the support for PySide2).

But I don't want to wait until that to merge the feature/sso branch into master.

Reply:
I will rename the package and move it to tank_vendor.